packages/dds/tree/src/feature-libraries/chunked-forest/codec/compressedEncode.ts

```diff
@@ -401,6 +403,75 @@ export class NestedArrayShape extends ShapeGeneric<EncodedChunkShape> implements
 	}
 }
 
+/**
+ * Encodes a field that supports incremental encoding. The chunks in this field will be encoded separately.
+ * The encoded data for this field will be an array of {@link ChunkReferenceId}s, one for each of its chunks.
+ */
+export class IncrementalFieldShape
```

Comment:
If you do the change suggested in https://github.com/microsoft/FluidFramework/pull/24949/files#r2232226768

then IncrementalFieldShape would most easily be expressed as a NestedArray of IncrementalChunks. The current implementation of NestedArray only supports Node encoders though, not NodesEncoders, so that would have some issues.

To avoid those problems, but keep the desired encoded format, you could keep this as a field encoder, but have it repor its encoded shape as a NestedArrayShape of IncrementalChunks.

To do that you would want an IncrementalChunk `Shape` implementation which this encoder could use, likely as a singleton like anyNodeEncoder or anyFieldEncoder